pkg/network/node/networkpolicy.go

```diff
@@ -494,6 +495,7 @@ func (np *networkPolicyPlugin) parseNetworkPolicy(npns *npNamespace, policy *net
 		// policy that only affects egress is, for our purposes, equivalent to one
 		// that affects ingress but does not select any pods.
 		npp.selectedIPs = []string{""}
+		npp.matchesAllIPs = true
```

Comment:
You need to keep your original change here too; `npp.selectedIPs = nil`

Reply:
I don't quite follow this, I'll ping you on slack to follow up.

Reply:
ok, followup from slack chat, the idea is that we wouldn't be looking at `npp.selectedIPs` when `npp.matchesAllIPs` is true. And, also, we shouldn't be putting an empty ip address in `npp.selectedIPs `ever (invalid, empty ip address)

Reply:
done